### DISCUSSION TITLE
How to display pdfplumber.display.PageImage in a for loop

### DISCUSSION THREAD

**Discussion by PatrykKoperski94** [jsvine/pdfplumber#472] [Q&A]
Hi,

I would like to output all pages in a pdf into the console using a for loop like so:

```
for page in enumerate(pdf.pages) :
     pdf.pages[page[0]].to_image()
```

However, everything works perfectly when I do it one by one 'manually' like so:
```
pdf.pages[0].to_image()
pdf.pages[1].to_image()
pdf.pages[2].to_image()
```

However, I notice that there is a small loading time with each image, and if I run the three lines above at once, only the last one gets displayed. Can someone explain what actually happens here?

Is there a way to display each image in the console using an iterative approach?

Thank you

**Comment by samkit-jain** [comment]
Hi @PatrykKoperski94 Assuming by display you refer to saving the page as an image, you may use the following
```python
import pdfplumber

pdf = pdfplumber.open("file.pdf")

for page in pdf.pages:
    im = page.to_image()
    im.save(f"{page.page_number}.png", format="PNG")
```

It would save images as `1.png`, `2.png`, ...
